include/cvs/common/configbase.hpp

```diff
@@ -129,14 +129,38 @@ enum class ConfigValueKind {
   WITH_DEFAULT_VALUE,
 };
 
+template <typename Subtype>
+std::optional<Subtype> get_value_from_ptree(const boost::property_tree::ptree& source, const std::string& name) {
+  constexpr bool is_vector_specialization = cvs::common::utils::is_specialization<Subtype, std::vector>::value;
+  static_assert(std::is_arithmetic_v<Subtype> || std::is_same_v<Subtype, std::string> || is_vector_specialization,
+                "Subtype must not be a arithmetic type or a std::string or a vector");
+  if constexpr (!is_vector_specialization) {
+    return utils::boostOptionalToStd(source.get_optional<Subtype>(name));
+  } else {
+    const auto&            object = source.get_child_optional(name);
+    std::optional<Subtype> result;
+    if (object && !object->empty() && object->data().empty()) {
+      result = Subtype();
+      for (const auto& item : object.get()) {
+        auto value_optional = item.second.template get_value_optional<Subtype::value_type>();
```

Comment:
Без `typename` компилятор не понимает, что `Subtype::value_type` это тип. 

```suggestion
        auto value_optional = item.second.template get_value_optional<typename Subtype::value_type>();
```
Или можно через `using` ему альяс объявить, там контекст однозначный.
```cpp
using ValueType = Subtype::value_type;
...
get_value_optional<ValueType>();
```